cw_infrastructure/cw_infrastructure/validators.py

```diff
@@ -0,0 +1,278 @@
+        return ERROR
+
+    scheme = match.group(1)
+    host = match.group(3)
+    port = match.group(4)
+    params = match.group(5)
+
+    if scheme != "sip":
+        error(name, "{} is not a SIP URI".format(value))
+        return ERROR
+
+    if params:
+        pdict = {}
+
+        for param in params.split(';'):
+            pmatch = re.match(r"([^=]+)=(.*)", param)
+            if pmatch:
+                pdict[pmatch.group(1)] = pmatch.group(2)
+            else:
+                pdict[param] = True
+
+        params = pdict
+    else:
+        params = {}
+
+    # Check whether a transport parameter is provided and is valid
+    transport = None
+
+    if 'transport' in params:
+        if params['transport'].lower() not in ('udp', 'tcp'):
+            error(name, ("{} is not a valid SIP "
+                         "transport").format(params['transport']))
+            return ERROR
+
+        else:
+            transport = params['transport']
+
+    if utils.is_ip_addr(host):
+        return OK
+
+    elif not utils.is_domain_name(host):
+        error(name, ("{} is neither an IP address or a valid domain "
+                     "name".format(host)))
+        return ERROR
+
+    elif port:
+        if utils.is_resolvable_domain_name(host):
+            return OK
+
+        else:
+            error(name, "{} is not resolvable".format(host))
+            return ERROR
+
+    elif transport:
+        srv = '_sip._{}.{}'.format(params['transport'], host)
+
+        if utils.is_srv_resolvable(srv):
+            return OK
+        else:
+            error(name, "{} is not a valid SRV record".format(srv))
+            return ERROR
+    else:
+
+        if utils.is_naptr_resolvable(host):
+            return OK
+        else:
+            error(name, "{} is not a valid NAPTR record".format(host))
+            return ERROR
+
+
+def diameter_realm_validator(name, value):
+    """Validate a config option that should be a diameter realm"""
+
+    # Realms should be valid domain names
+    if not utils.is_domain_name(value):
+        error(name, "{} is not a valid realm".format(value))
+        return ERROR
+
+    if not utils.is_srv_resolvable('_diameter._tcp.' + value):
+        error(name, (
+            '_diameter._tcp.{} does not resolve to any SRV '
+            'records'.format(value)))
+        return ERROR
+
+    return OK
+
+
+def ip_or_domain_name_with_port_validator(name, value):
+    """Validate a config option that should be a IP address or domain name,
+    followed by a port"""
+    match = re.match(r"^(.*):(\d+)$", value)
+    if not match:
+        error(name, "{} does not contain a port".format(value))
+        return ERROR
+
+    stem = match.group(1)
+    port = match.group(2)
+
+    if int(port) >= 2**16:
+        error(name, "The port value ({}) is too large".format(port))
+        return ERROR
+
+    if (stem[0] == '[') and (stem[-1] == ']'):
+        ip = stem[1:-1]
+        if ip_version(ip) == 6:
```

Comment:
I think this should be `utils.ip_version(ip)`.

Has flake8 been run on this code? I think it would normally pick up this kind of bug.